core/src/Kokkos_HostSpace.hpp

```diff
@@ -261,13 +261,27 @@ class SharedAllocationRecord<Kokkos::HostSpace, void>
       ;
   SharedAllocationRecord() = default;
 
+  // This constructor does not forward to the one without exec_space arg
+  // in order to work around https://github.com/kokkos/kokkos/issues/5258
+  // This constructor is templated so I can't just put it into the cpp file
+  // like the other constructor.
   template <typename ExecutionSpace>
   SharedAllocationRecord(
       const ExecutionSpace& /* exec_space*/, const Kokkos::HostSpace& arg_space,
       const std::string& arg_label, const size_t arg_alloc_size,
       const RecordBase::function_type arg_dealloc = &deallocate)
-      : SharedAllocationRecord(arg_space, arg_label, arg_alloc_size,
-                               arg_dealloc) {}
```

Comment:
Did you try this more explicit syntax to get the forwarding constructor?
```
 : SharedAllocationRecord<Kokkos::HostSpace, void>(arg_space, arg_label, arg_alloc_size,
                               arg_dealloc) {}
```

Reply:
yes